### PR TITLE
Allow to set own redis driver

### PR DESCRIPTION
**Pull request by legendik**
Signed-off-by: Marek Šneberger <marek@sneberger.cz>

We need to set own Redis driver, which will work with Sentinel.

### CODE REVIEW

File: src/Kdyby/Redis/DI/RedisExtension.php [hunk 1]
```diff
@@ -68,11 +68,10 @@ public function loadConfiguration()
 		$builder = $this->getContainerBuilder();
 		$config = self::fixClientConfig($this->getConfig($this->defaults + $this->clientDefaults));
 
-		$this->buildClient(NULL, $config);
-
 		$builder->addDefinition($this->prefix('driver'))
-			->setClass(class_exists('Redis') ? 'Kdyby\Redis\Driver\PhpRedisDriver' : 'Kdyby\Redis\IRedisDriver')
-			->setFactory($this->prefix('@client') . '::getDriver');
+			->setClass(Kdyby\Redis\Driver\PhpRedisDriver::class);
```
Comment (enumag) [inline src/Kdyby/Redis/DI/RedisExtension.php]: Return the class_exists. I think it's there to allow this extention to be installed on environment where Redis class is not available.

File: src/Kdyby/Redis/RedisClient.php [hunk 2]
```diff
@@ -287,7 +297,7 @@ public function connect()
 			throw new MissingExtensionException("Please install and enable the redis extension. \nhttps://github.com/nicolasff/phpredis/");
 		}
 
-		if (!$this->driver) {
+		if ($this->driver === NULL) {
```
Comment (enumag) [inline src/Kdyby/Redis/RedisClient.php]: revert this

File: tests/KdybyTests/Redis/files/NullDriver.php [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+	public function script($command, $script = NULL) {}
+
+	public function clearLastError(){}
+
+	public function close(){}
+
+	public function evalsha($scriptSha, $argsArray = [], $numKeys = 0){}
+
+	public function getLastError(){}
```
Comment (enumag) [inline tests/KdybyTests/Redis/files/NullDriver.php]: fix coding style in this file

**Comment by enumag** [comment]
@martinvenus What do you think about this?

**Comment by martinvenus** [comment]
@enumag Sorry for late reply. I have to say that I'm not sure about this code. I'm afraid that I'm not so experienced to be able to say if it's OK.

I also asked one of my colleague to check it it's OK, but he isn't also sure.

**Comment by legendik** [comment]
Alt Text

**Pull request closed without merging**